fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/PermissiveAvailability.java

```diff
@@ -26,36 +25,36 @@
     /**
      * Constructor.
      *
-     * @param tableName The name of the data source
-     * @param columns A set of columns associated with the data source
-     * @param metadataService A service containing the data source segment data
+     * @param tableName  The name of the data source
+     * @param metadataService  A service containing the data source segment data
      */
     public PermissiveAvailability(
             @NotNull TableName tableName,
-            @NotNull Set<Column> columns,
             @NotNull DataSourceMetadataService metadataService
     ) {
-        super(tableName, columns, metadataService);
+        super(tableName, metadataService);
     }
 
     /**
      * Returns union of all available intervals.
      * <p>
      * This is different from its parent's
      * {@link
-     * com.yahoo.bard.webservice.table.availability.ConcreteAvailability#getAvailableIntervals(DataSourceConstraint)};
+     * com.yahoo.bard.webservice.table.availability.ConcreteAvailability#getAvailableIntervals
+     * (PhysicalDataSourceConstraint)};
      * Instead of returning the intersection of all available intervals, this method returns the union of them.
      *
-     * @param ignoredConstraints  Data constraint containing columns and api filters. Constrains are ignored, because
+     * @param constraint  Data constraint containing columns and api filters. Constrains are ignored, because
      * <tt>PermissiveAvailability</tt> returns as much available intervals as possible by, for example, allowing
      * missing intervals and returning unions of available intervals
      *
      * @return the union of all available intervals
      */
     @Override
-    public SimplifiedIntervalList getAvailableIntervals(DataSourceConstraint ignoredConstraints) {
-        Map<String, List<Interval>> allAvailableIntervals = getAvailableIntervalsByTable();
-        return getColumnNames().stream()
+    public SimplifiedIntervalList getAvailableIntervals(PhysicalDataSourceConstraint constraint) {
+        Map<String, List<Interval>> allAvailableIntervals = getAllAvailableIntervals();
+
+        return constraint.getAllColumnPhysicalNames().stream()
```

Comment:
I did a change here that makes it the same I think, [here](https://github.com/yahoo/fili/pull/226/files#diff-2cf930e01f4320995434024e1678779cR56)